atomics/include/desul/atomics/Compare_Exchange_GCC.hpp

```diff
@@ -17,58 +17,74 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #endif
 namespace desul {
 
-#if defined(__clang__) && (__clang_major__>=7) && !defined(__APPLE__)
+namespace Impl {
+template <class T>
+struct atomic_exchange_available_gcc {
+  constexpr static bool value =
+#ifndef DESUL_HAVE_LIBATOMIC
```

Comment:
Where is this defined? Is this supposed to be defined outside of `desul`?

Reply:
Outside, compiler option `-D DESUL_HAVE_LIBATOMIC=1` or in code before including desul.  (Just describing the status quo, not saying it is a great design.)